internal/util/context.go

```diff
@@ -19,8 +19,15 @@ func ContextWithLogger(ctx context.Context, logger *logrus.Entry) context.Contex
 
 // LoggerFromContext returns the logger from the context.
 func LoggerFromContext(ctx context.Context) *logrus.Entry {
-	logger := ctx.Value(loggerKey).(*logrus.Entry)
-	if logger == nil {
+	value := ctx.Value(loggerKey)
+	if value == nil {
+		logrus.Warnf("nil logger in context, creating a new one")
+		return logrus.NewEntry(logrus.New())
+	}
+
+	logger, ok := value.(*logrus.Entry)
+	if !ok {
+		logrus.Warnf("received wrong type of logger (%T)", logger)
 		logger = logrus.NewEntry(logrus.New())
```

Comment:
you should learn by now, this is code repetition. you have 2 branches that do the same thing.

for example, if you tried it yourself instead of directly accepting my suggestion you'd find that nil actually works with `v, ok := ...(type)` assertion just fine.

https://play.golang.org/p/Nt2AyjMiqwG